src/sentry/api/endpoints/organization_access_request_details.py

```diff
@@ -71,9 +71,11 @@ def _can_access(self, request: Request, access_request):
 
     def get(self, request: Request, organization) -> Response:
         """
-        Get list of requests to join org/team
-
+        Get a list of requests to join org/team.
+        If any requests are outdated, take this opportunity to prune them.
         """
+        OrganizationAccessRequest.objects.prune_outdated_team_requests(organization=organization)
```

Comment:
I would not normally say no to this kind of change but now after [inc-886](https://www.notion.so/sentry/INC-886-Members-being-removed-without-audit-log-1128b10e4b5d813492a4d222769d5555) I highly recommend doing a celery task. Basically because if there's an issue in this code finding it out takes us a lot of time because the deletion happens at different time for different orgs and there are no audit log trail to let the org members know what's happening.

Reply:
Do you have a recommendation for how to trigger the tasks? Should this just be something run on every organization every X timeframe? I could try to nail down how many production rows could be pruned this way so we have an idea.

Reply:
@sentaur-athena I've rewritten this to just return filtered entries to the frontend. This means that we're still going to have these rows in our database, but they won't be provided to the frontend if they're redundant.

Reply:
Yeah I see the problem. Is this possible to prevent from happening in future completely? Then we would have to run the task only once. What flows do we need to cover so this case doesn't happen?

Reply:
I did look into that but I'm not sure of how many places we'd have to make changes. It seems the only the access request gets deleted is if it approved, but any other way of joining a team will have no affect on the request. Manually being added, SCIM, and enabling open enrollment were the few methods I noted.

For now I'll merge this as is to prevent the unexpected behavior in the app, thanks for the review